### ISSUE TITLE
Doubtful error 'UnnecessaryVarAnnotation' in the included file

### ISSUE THREAD

**Issue by kastaneda** [vimeo/psalm#11605]
I used to use `/** @var SomeClassName $this */` in template files, to make IDE autocomplete working well and to keep a note for myself. But Psalm finds this annotation unnecessary.

Isn't it too ambiguous, when it comes to standalone files?

**index.php**

```php
<?php

final class HelloPage
{
    public function __construct(
        public string $name,
    ) {
    }

    public function render(): void
    {
        require __DIR__ . '/template.php';
    }
}

$page = new HelloPage('World');
$page->render();
```

**template.php**

```php
<?php /** @var \HelloPage $this */ ?>
<html>
    <head>
        <title>UnnecessaryVarAnnotation</title>
    </head>
    <body>
        <h1>Hello, <?php echo htmlspecialchars($this->name); ?>!</h1>
        <p><a href="https://psalm.dev/212">Really?</a></p>
    </body>
</html>
```

**Comment by kkmuffme** [comment]
Which IDE are you using that is unable to correctly infer this?

If this affects "template.php" (or /template/ paths) specifically, you can suppress just UnnecessaryVarAnnotation issue as explained here https://github.com/vimeo/psalm/issues/9774#issuecomment-3998693577

**Comment by kastaneda** [comment]
It was Visual Studio Code. On the day when the issue was reported (and probably later), the IDE was unable to correctly complete `$this->` followed by Ctrl-Space without specifying class for `$this`.

You are right, I can suppress these error messages for the specific path. Actually, I do that. But I believe there might be a better solution.

**Comment by kkmuffme** [comment]
If you try HelloPage without the FQCN qualifier/backslash maybe VSCode has an issue with that?

Anyway, this isn't an issue in psalm, but a bug in VSCode, report it there?
